core/compiler.cpp

```diff
@@ -180,8 +189,32 @@ partitioning::GraphAndMapping BuildHybridGraph(
             true);
 
         seg_block.update_graph(temp_g);
+      } else {
+        num_torch_segments++;
+
+        // If full compilation is expected, ensure that all operators in Torch blocks are
+        // for collections processing
+        if (expect_full_compilation) {
+          for (auto torch_node : seg_block.block()->nodes()) {
+            if (partitioning::CollectionSchemas.find(torch_node->kind().toQualString()) ==
+                partitioning::CollectionSchemas.end()) {
+              LOG_WARNING(
+                  "Full compilation specified but node " << torch_node->kind().toQualString()
+                                                         << " was executed in Torch.");
+            }
+          }
+        }
       }
     }
+
+    // If full compilation is expected, cannot have more than 2 Torch segments
+    // (one for preprocessing inputs, one for post-processing outputs) and 1 TRT segment
+    if (expect_full_compilation && !(num_torch_segments <= 2 && num_trt_segments == 1)) {
+      LOG_WARNING(
+          "Full compilation specified but number of torch segments was "
+          << num_torch_segments << " and number of trt segments was " << num_trt_segments
+          << ". Was expecting at most 2 Torch segments and 1 TRT segment.");
```

Comment:
Similarly to the above comment, this could alternatively be made to halt compilation, since it indicates the model is no longer being "fully-compiled" as there are multiple TRT engine blocks.

Reply:
This should be an error if it does not meet our criteria for full compilation